src/coreclr/vm/jitinterface.cpp

```diff
@@ -9323,6 +9323,532 @@ CorInfoTypeWithMod CEEInfo::getArgType (
+
+    Module* pModule = GetModule(sig->scope);
+
+    CorElementType type = ptr.PeekElemTypeClosed(pModule, &typeContext);
+
+    TypeHandle typeHnd = TypeHandle();
+    switch (type) {
+      case ELEMENT_TYPE_VAR :
+      case ELEMENT_TYPE_MVAR :
+      case ELEMENT_TYPE_VALUETYPE :
+      case ELEMENT_TYPE_TYPEDBYREF :
+      case ELEMENT_TYPE_INTERNAL :
+      {
+            typeHnd = ptr.GetTypeHandleThrowing(pModule, &typeContext);
+            _ASSERTE(!typeHnd.IsNull());
+
+            CorElementType normType = typeHnd.GetInternalCorElementType();
+
+            // if we are looking up a value class, don't morph it to a refernece type
+            // (This can only happen in illegal IL)
+            if (!CorTypeInfo::IsObjRef(normType) || type != ELEMENT_TYPE_VALUETYPE)
+            {
+                type = normType;
+            }
+            if ((pFloatFieldFlags != NULL) && (type == ELEMENT_TYPE_VALUETYPE))
+            {
+                *pFloatFieldFlags = getFieldTypeByHnd((CORINFO_CLASS_HANDLE)typeHnd.AsTAddr());
+            }
+        }
+        break;
+
+    case ELEMENT_TYPE_PTR:
+        // Load the type eagerly under debugger to make the eval work
+        if (!isVerifyOnly() && CORDisableJITOptimizations(pModule->GetDebuggerInfoBits()))
+        {
+            // NOTE: in some IJW cases, when the type pointed at is unmanaged,
+            // the GetTypeHandle may fail, because there is no TypeDef for such type.
+            // Usage of GetTypeHandleThrowing would lead to class load exception
+            TypeHandle thPtr = ptr.GetTypeHandleNT(pModule, &typeContext);
+            if(!thPtr.IsNull())
+            {
+                classMustBeLoadedBeforeCodeIsRun(CORINFO_CLASS_HANDLE(thPtr.AsPtr()));
+            }
+        }
+        break;
+
+    case ELEMENT_TYPE_VOID:
+        // void is not valid in local sigs
+        if (sig->flags & CORINFO_SIGFLAG_IS_LOCAL_SIG)
+            COMPlusThrowHR(COR_E_INVALIDPROGRAM);
+        break;
+
+    case ELEMENT_TYPE_END:
+           COMPlusThrowHR(COR_E_BADIMAGEFORMAT);
+        break;
+
+    default:
+        break;
+    }
+
+    result = CorInfoTypeWithMod(result | CEEInfo::asCorInfoType(type, typeHnd, vcTypeRet));
+    EE_TO_JIT_TRANSITION();
+
+    return result;
+}
+
+// NOTE:
+// Although this is only used for LoongArch64-ABI now,
+// maybe it can be used for other architecture for getting ABI-info
+// between JIT/EE if the ABI is similar.
+//
+// Now the implementation is only focused on the float fields info,
+// while a struct-arg has no more than two fiels and total size is no larger than two-pointer-size.
+// These depends on the platform's ABI rules.
+//
+// The returned value's encoding details whether a struct-arg using float regitsters:
+// returnd `0` means not using the float register(s).
+//
+// the lowest four bits denoting the floating-point info:
+//   bit_0: `1` means there is only one float or double field within the struct.
```

Comment:
We should have named constants or enum for these.

Reply:
> We should have named constants or enum for these.

Thanks. I agree with you!
How about define named constants macro-defines ?

Reply:
OK, thanks.